src/ufixed.jl

```diff
@@ -35,25 +37,38 @@ zero{T,f}(::Type{UFixed{T,f}}) = UFixed{T,f}(zero(T),0)
     f = 2^nbitsfrac(T)-1
     :( T($f,0) )
 end
+one{f}(::Type{UFixed{UInt128,f}}) = UFixed{UInt128,f}(UInt128(2)^f-1, 0)
 zero(x::UFixed) = zero(typeof(x))
  one(x::UFixed) =  one(typeof(x))
 rawone(v) = reinterpret(one(v))
 
 # Conversions
 convert{T<:UFixed}(::Type{T}, x::T) = x
-convert{T1<:UFixed}(::Type{T1}, x::UFixed) = reinterpret(T1, round(rawtype(T1), (rawone(T1)/rawone(x))*reinterpret(x)))
+function convert{T<:UFixed}(::Type{T}, x::UFixed)
+    y = round((rawone(T)/rawone(x))*reinterpret(x))
+    (0 <= y) & (y <= typemax(rawtype(T))) || throw_converterror(T, x)
```

Comment:
Performance. Currently `a <= x <= b` gets lowered to `(a <= x) && (x <= b)` rather than `(a <= x) & (x <= b)`, which means there are two branches rather than one.

This is something that should probably be changed in Julia, but neither I nor anyone else has gotten to it yet.